### ISSUE TITLE
[BUG] No `NotFittedError` is raised 

### ISSUE THREAD

**Issue by mloning** [sktime/sktime-dl#37]
** Bug description
sklearn's `check_is_fitted()` by default checks if any attribute with a trailing underscore is present, if so, it assumes the estimator has been fitted. We initialise `is_fitted_=False` in the constructor, so the checks pass, ignoring the value of `is_fitted_` simply by finding an attribute with a trailing underscore.

**To Reproduce**
```python
from sktime_dl.deeplearning import MLPRegressor
from sktime.datasets import load_gunpoint

X_train, y_train = load_gunpoint("TRAIN", return_X_y=True)
r = MLPRegressor()
r.predict(X_train)
```

**Expected behavior**
raise `NotFittedError`, instead we get 
`AttributeError: 'NoneType' object has no attribute 'predict'`

**Proposed solution**
- [ ] I'm not a big fan of scikit-learn's solution using the attributes-with-trailing-underscore convention. Should we simply implement a `check_is_fitted()` method on the base class which checks if `_is_fitted=True`? 
- [ ] add unit test

**Comment by James-Large** [comment]
Yeah let's just replace it. Without looking too deep into it, can't think how it's even generating that error or passing the check_is_fitted.

Will pr this later, and add the test to test_networkcapabilities

**Comment by mloning** [comment]
I've updated the description a little bit to explain what's going on.

**Comment by James-Large** [comment]
> We initialise is_fitted_=False in the constructor

Ah, oops

**Comment by James-Large** [comment]
The main issue/bug here has been fixed within pr #39, however this topic is to be generalised across to base sktime in the future and the method updated here on -dl too, #41
